src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -1458,38 +1458,51 @@ public void readExternal(DataInputStream dis, PrototypeFactory pf) throws IOExce
+    }
+
     public void initialize(boolean newInstance, boolean isCompletedInstance,
                            InstanceInitializationFactory factory) {
-        initialize(newInstance, isCompletedInstance, factory, null, false);
+        initialize(newInstance, isCompletedInstance, factory, null, false, true);
     }
 
-    public void initialize(boolean newInstance, InstanceInitializationFactory factory) {
-        initialize(newInstance, false, factory, null, false);
+    public void initialize(boolean newInstance, InstanceInitializationFactory factory, String locale, boolean isReadOnly) {
+        initialize(newInstance, isCompletedInstance, factory, locale, isReadOnly, true);
     }
 
-    public void initialize(boolean newInstance, InstanceInitializationFactory factory, String locale,
-                           boolean isReadOnly) {
-        initialize(newInstance, false, factory, locale, isReadOnly);
+    public void initialize(boolean newInstance, InstanceInitializationFactory factory, String locale) {
+        initialize(newInstance, false, factory, locale, false, true);
     }
 
+    public void initialize(boolean newInstance,
+                           boolean isCompletedInstance,
+                           InstanceInitializationFactory factory,
+                           String locale) {
+        initialize(newInstance, isCompletedInstance, factory, locale, false, true);
+    }
     /**
      * meant to be called after deserialization and initialization of handlers
      *
      * @param newInstance true if the form is to be used for a new entry interaction,
      *                    false if it is using an existing IDataModel
+     * @param isCompletedInstance true if this is an already completed instance we are editing
+     *                            (presumably in HQ) - so don't fire end of form event.
+     * @param factory instance factory containing references to external data sources
      * @param locale The default locale in the current environment, if provided. Can be null
      *               to rely on the form's internal default.
+     * @param isReadOnly true if this form has already been saved and so should not be re-processed
+     *
+     * @param initializeTriggerables true if this form is being reloaded from an incomplete form
+     *                                and so we should re-process triggerables to account for
+     *                                changes to user databases
      */
-    public void initialize(boolean newInstance, boolean isCompletedInstance,
-                           InstanceInitializationFactory factory, String locale, boolean isReadOnly) {
+    public void initialize(boolean newInstance,
+                           boolean isCompletedInstance,
+                           InstanceInitializationFactory factory,
+                           String locale,
+                           boolean isReadOnly,
+                           boolean initializeTriggerables) {
```

Comment:
@wpride Hey so this variable re-naming definitely makes more sense, but it made me realize that there's 1 more thing that should change here. The `isReadOnly` argument was added (by me) fairly recently to serve essentially the same role that `initializeTriggerables` is now playing. You'll notice that the only usage of `isReadOnly` is in that one line below to control initializing triggerables. So it would be cleaner and a more accurate representation of what's going on here if you now removed the `isReadOnly` arg entirely, and then passed in `false` for `initializeTriggerables` in all places where we were previously passing `true` for `isReadOnly`. (If that wasn't totally clear, I can also just do this refactor if you want me to; let me know).